This was accepted upstream. Should be obsolete in next version after 1.12.

http://lists.gnu.org/archive/html/bug-wget/2010-01/msg00007.html

--- src/init.c.orig	2009-11-23 10:03:09.664171000 -0600
+++ src/init.c	2009-11-23 10:02:10.556336000 -0600
@@ -37,6 +37,15 @@
 #endif
 #include <string.h>
 #include <errno.h>
+#include <limits.h>
+/* not all systems provide PATH_MAX in limits.h */
+#ifndef PATH_MAX
+# include <sys/param.h>
+# ifndef PATH_MAX
+#  define PATH_MAX MAXPATHLEN
+# endif
+#endif
+
 
 #ifdef HAVE_PWD_H
 # include <pwd.h>
